src/strategies/circuit-breaker.strategy.ts

```diff
@@ -53,6 +57,18 @@ export class CircuitBreakerStrategy extends Strategy<CircuitBreakerOptions> {
 				const failuresPercentage = () =>
 					(state.failuresCount / this.options.requestVolumeThreshold) * 100;
 
+				if (this.options.rollingWindowInMilliseconds) {
+					const rollingWindowHasExpired = () =>
+						Date.now() >=
+						state.lastRequestTimeMs + this.options.rollingWindowInMilliseconds;
+					if (rollingWindowHasExpired()) {
+						state.failuresCount = 0;
+						state.succeedsCount = 0;
+						state.openedAt = 0;
+						state.lastRequestTimeMs = Date.now();
```

Comment:
Why we set last request only when rolling window has expired not every request? If it`s has no meaning - it`ll be merged

Reply:
You're right it should be every request. Made the change here: [d309e91](https://github.com/SocketSomeone/nestjs-resilience/pull/320/commits/d309e913775d06f6b67153a76474b21c7cc610ad)